src/server.ts

```diff
@@ -26,120 +33,170 @@ export class MyAgent extends Agent<Env> {
     apiKey: this.env.GROQ_API_KEY,
```

Comment:
2 lines above, I didn't notice you had a new class `AIChatAgent` until it was too late. It probably would have saved me a lot of work but whatevs